tests/e2e/manifest/setup.sh

```
@@ -0,0 +1,67 @@
+  local template_file="$1"
+  local target_namespace="$2"
+  
+  echo "Applying template: $(basename "$template_file")"
+  
+  # Substitute all variables and apply
+  sed -e "s/\${NAMESPACE}/$target_namespace/g" "$template_file" | kubectl apply -f -
+}
+
+# 1. Apply target deployment
+apply_template "$MANIFEST_DIR/test-template/target-deployment.yaml" "$NAMESPACE"
+kubectl wait --for=condition=Ready pods -l app=target-deployment -n $NAMESPACE
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)